src/components/Checkbox/BaseCheckbox.js

```diff
@@ -26,14 +30,15 @@ const propTypes = {
     /** Additional styles to add to checkbox button */
     style: stylePropTypes,
 
-    /** Callback that is called when mousedown is triggered. */
-    onMouseDown: PropTypes.func,
-
     /** A ref to forward to the Pressable */
     forwardedRef: PropTypes.oneOfType([
         PropTypes.func,
         PropTypes.shape({current: PropTypes.instanceOf(React.Component)}),
     ]),
+
+    onPress: requiredPropsCheck,
+
```

Comment:
```suggestion
/** Callback that is called when mousedown is triggered. */
```

Please add those comments back